crates/pixi_glob/src/glob_set.rs

```diff
@@ -140,4 +173,84 @@ mod tests {
         expected.sort();
         assert_eq!(filtered_files, expected);
     }
+
+    #[test]
+    fn test_filters_with_relatives_globs() {
+        // In this test we want to make sure that when globbing over
+        // patterns that contains semantic relative path, like
+        // ../pixi.toml or ../sources/*.toml, we are able to
+        // distinguish between glob and just a semantic path.
+        let temp_dir = tempdir().unwrap();
+        let root_path = temp_dir.path();
+
+        let temp_path_as_root = temp_dir.path().join("somewhere_inside");
+        fs_err::create_dir(&temp_path_as_root).unwrap();
+
+        // Create files and directories
+        File::create(root_path.join("include1.txt")).unwrap();
+        File::create(root_path.join("include2.log")).unwrap();
+        File::create(root_path.join("exclude.txt")).unwrap();
```

Comment:
```suggestion
```

I think it's better to focus the test a bit more.
Let's only include that one cpp file we care about and make sure it's included